envoy/access_log/access_log_config.h

```diff
@@ -84,5 +84,42 @@ template <class Context> class AccessLogInstanceFactoryBase : public Config::Typ
   const std::string category_;
 };
 
+/**
+ * Implemented for each AccessLog::Instance and registered via Registry::registerFactory or the
+ * convenience class RegisterFactory.
+ */
+class AccessLogInstanceFactory : public Config::TypedFactory {
```

Comment:
I'm a bit unclear why this new Factory is being added as part of this PR. It's probably obvious, but from the PR description, I don't understand why.

Reply:
Because this is only 1/3 of a complete refactoring, so it maybe a little confusing.

This `AccessLogInstanceFactory` is moved from `/envoy/server/access_log_config.h` to here and will be replaced by the `AccessLogInstanceFactoryBase<HttpLogContext>` finally.

I did this moving in this patch, then in the new sub PRs, we needn't to update logger implementations again. This will make the PR more clean and easy to review (but seems it still bring some confusions).